### ISSUE TITLE
Desired feature: Qiqqa should provide intelligent resource search

### ISSUE THREAD

**Issue by raindropsfromsky** [jimmejardine/qiqqa-open-source#184]
During the **knowledge discovery**, the first important step is to search for potentially good source pdf files.

But in the whole description, Qiqqa seems to skip this step altogether. It simply assumes that we somehow have all the pdf files we need, and we can start analyzing their content to find similar material (by matching theme/tag+expedition), and interlinking the core ideas (brainstorming).

Therefore Qiqqa should have a good mechanism to search the online databases and find the best-matching sources.

Here, two reference managers are worth benchmarking: **Zotero** and **Citavi**.
- In Zotero, we have to use a browser to search research databases and book catalogs, and then use the Zotero extension to capture the list into Zotero. It can process all [online databases](https://paperpile.com/g/academic-research-databases/), and also catalogs like Amazon, etc.
- Citavi has a similar browser extension. But it also has a much more sophisticated [metadata search](https://www1.citavi.com/sub/manual5/en/101_searching_research_databases_with_citavi.html), which can connect to any desired specialized database, and find the most recommended research papers from them.
- The Docear uses yet another trick: A document recommender system that uses multi-graph and article ranking to find the most relevant documents.

In first two software, the search-result yields a list of articles/books, and their metadata (in most cases), but not the actual pdf files. These have to be downloaded separately (some have to be purchased). In Docear, you can download the pdf with one click.

**Desired:** 
Provide tools that help the researcher with Internet search for the most suitable resources. Otherwise, if the researcher starts with junk stuff, Qiqqa cannot help him: It will always be GIGO.

**Comment by GerHobbelt** [comment]
Ehm, I'm confused. Did you check the BibTeX Sniffer? I use that one to search the Internet for related PDFs and add them to the library. The Sniffer will start using the title of the current document and submits that to Google Scholar or any of the other search engines (available in the browser tabs in the Sniffer; you can configure the set of search engines displayed in there by clicking the configuration icon nearby in the same Sniffer window)

**Comment by raindropsfromsky** [comment]
Yes, I do know about BibTex sniffer, but it is pitched as a tool to find the metadata for a given file. It is a powerful tool, indeed, and a major timesaver.

But it can NOT be used as a tool to find the most matching documents in Google Scholar, because it matches the title, and nothing else. So it is not likely to suggest articles and books with matching content.

Besides, the capabilities of the three apps I mentioned are on a completely different plane.

In fact, this is how I build my document base in the first place: First visit some of my favorite websites (including Google Scholar) and search for all the keywords and keyphrases one by one. Then I look through the results, and download anything that looks interesting.

Thus Zotero and Citavi just facilitate that same process, without trying to pick the most suitable docs for me. Only Docear has a built-in analyzer (I haven't tried how clever it is).

But in case of Qiqqa, it already does the analysis to find the documents that I should read next.

What I need to do is to automate my search process, based on the keywords/phrases I want.
It need not be a much intelligent tool.

BTW I think (not sure) that Google Scholar as a single source of information may not be adequate; given that there are so many [research databases](https://paperpile.com/g/academic-research-databases/) out there. I am sure that those databases would have hidden treasures that Google Scholar cannot access.

**Comment by raindropsfromsky** [comment]
Let me illustrate this with an example:

I am working on **lake rejuvenation**, which can be broken down in various interrelated concepts. I need articles that deal with all of those concepts. Thus, what I really need is a [concept search](https://www.researchgate.net/publication/220853970_Concept_Search) engine.

Thus, for a given subject, I first think of its [taxonomy](https://211taxonomy.org/search/related) and then use those keywords for searching.

The articles will have nothing common in their title.

I do not know if there is a universal taxonomy database like this. But if there is one, then Qiqqa's search engine can use it to find all the related concepts and search for them.

For example, if I am looking for "marriage", then I could be doing the research on [types of marriages](https://en.wikipedia.org/wiki/Types_of_marriages) or trying to create a todo list for my cousin's wedding as per Hindu customs. 

Atv the beginning. no system can predict the direction of my search from just a single word "marriage" . But when I choose more and more keywords/phrases, then it will know.

Conversely, suggesting more books based on the content-analysis of the current book seems to be a waste of time to me: The user should manually givve more specific hint by giving more keywords/phrases to work on.
